### PR TITLE
bugfix: Make scrollToEnd to be more reliable

### LINKED ISSUE #4497: Grid.scrollToEnd does not work if called before attaching
### Description

scrollToEnd method fails if called before the component is attached DOM. Quite typical use case: building a log view where grid is initialised in the constructor.

### Expected outcome

scrollToEnd should work

### Minimal reproducible example

```
@Route
public class ScrollToEndView extends VerticalLayout {
    public ScrollToEndView() {
        Grid<String> grid = new ProperGrid();
        grid.addColumn(s -> s);
        grid.setItems(IntStream.range(0,1000).mapToObj(i -> "" + i).collect(Collectors.toList()));
        grid.scrollToEnd();
        add(grid);
    }
}
```

### Steps to reproduce

1. Open view

See that grid is not scrolled to end

### Environment

Vaadin: 23.3.2
Flow: 23.3.2
Java: Eclipse Adoptium 19.0.1
OS: aarch64 Mac OS X 13.0.1
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
Live reload: Java active (Spring Boot Devtools): Front end active

### Browsers

_No response_

### PR DESCRIPTION
**Pull request by mstahv**
## Description

Fixes #4497

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

### CODE REVIEW

**Comment by mstahv** [comment]
Note that the current implementation seems very hacky and inefficient (causes extra roundtrips to server) to me, but with this change the method becomes at least functional for common usage.

**Comment by czp13** [comment]
Is this related to the scrollToIndex() GridTree problem? It would be amazing if this could be fixed 🙏

**Comment by mstahv** [comment]
It doesn't look like that. The scrollToEnd seems to have totally custom implementation. Do you have a link to the original issue in TreeGrid that you are facing?

**Comment by tomivirkki** [comment]
Making `scrollToIndex` work reliably with `TreeGrid` would require some new API, see [this discussion](https://github.com/vaadin/flow-components/issues/1016#issuecomment-829952022)

**Comment by czp13** [comment]
@mstahv: There are more tickets I assume, but I am aware of these ones:
- https://github.com/vaadin/flow-components/issues/1016
- https://github.com/vaadin/flow-components/issues/3505
Related to scrollToIndex/TreeGrid problems, 

Also, this same customer, who reported the above using extensively the TreeGrid, and experienced these ones on TreeGrid:
- https://github.com/vaadin/flow-components/issues/3355
- https://github.com/vaadin/flow-components/issues/1134

@tomivirkki: It would be quite nice to tackle some of these too with the new API.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java [hunk 1]
```diff
@@ -4416,9 +4417,9 @@ public void scrollToStart() {
      * Scrolls to the last data row of the grid.
      */
     public void scrollToEnd() {
-        getUI().ifPresent(
-                ui -> ui.beforeClientResponse(this, ctx -> getElement()
-                        .executeJs("this.scrollToIndex(this._effectiveSize)")));
+        getUI().orElseGet(UI::getCurrent).beforeClientResponse(this,
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java:4420]: How about using `getElement().getNode().runWhenAttached` to run the code when the component is actually added to a UI? There's a few other places in the component that use this approach.

**Review by sissbruecker — COMMENTED** [review]
Sorry for the late review. Not really happy with the `setTimeout` solution but I also couldn't find any alternatives.

Could you please add an integration test as mentioned in the contribution guidelines? `GridScrollToPage` would be a good place to start.

**Comment by yuriy-fix** [comment]
Closing the PR as stale as the last review comments have not been addressed in over a month. Please don't hesitate to open a new PR if you want to continue with this contribution. Thanks!

**Pull request closed without merging**